airbyte-webapp/src/hooks/services/AppMonitoringService/AppMonitoringService.tsx

```diff
@@ -10,7 +10,7 @@ const appMonitoringContext = createContext<AppMonitoringServiceProviderValue | n
  * These methods are particularly useful for tracking when unexpected or edge-case conditions
  * are encountered in production.
  */
-interface AppMonitoringServiceProviderValue {
+export interface AppMonitoringServiceProviderValue {
```

Comment:
This fixes the brittleness of manual redefining the tracking function types, which was my main practical issue before; love that. No problem merging this as-is; but FWIW, I have some thoughts about how to paint the bike shed an _even better_ color below :joy::

`AppMonitoringServiceProviderValue["trackAction"]` isn't the most ergonomic type signature to remember or type out; it makes the fact that `useAppMonitoringService` is backed by a context provider part of its public API, when I think it's really an implementation detail to be encapsulated. I think it would be a bit nicer if we split out named function types for each of the values, like
```ts
export type TrackActionFn = (actionCode: AppActionCodes, context?: Record<string, unknown>) => void;
export t pe TrackErrorFn = (error: Error, context?: Record<string, unknown>) => void;

// this could also be exported if another piece of code needs to be concerned with the full set of tracking options, but that seems like exactly the role of AppMonitoringService.tsx itself
interface AppMonitoringServiceProviderValue {
  trackAction: TrackActionFn;
  trackError: TrackErrorFn;
}
```
That way, the type defs and imports needed to addjust one of the tracking functions to a component* looks like
```ts
import { useAppMonitoringService, TrackActionFn } from "hooks/services/AppMonitoringService";

// ...

interface SpeakeasyErrorBoundaryProps {
  trackError: TrackActionFn;
}
```
which _feels_ easier to me.

\* I suspected we mainly used one or the other tracking function, rather than both, but I got curious and checked. Right now there is no application code using both in the same place:
```
┌ ~/c/airbyte/airbyte-webapp/src (master|✔) 🌓 885507f389
└➣ for f in $(rg -l trackAction); do rg -H trackError $f; done
hooks/services/AppMonitoringService/AppMonitoringService.tsx
21:  trackError: (error: Error, context?: Record<string, unknown>) => void;
46:  const trackError = (error: Error, context?: Record<string, unknown>) => {
48:      console.debug(`trackError() failed because RUM is not initialized. \n`, error);
55:  return <appMonitoringContext.Provider value={{ trackAction, trackError }}>{children}</appMonitoringContext.Provider>;
hooks/services/AppMonitoringService/__mocks__/AppMonitoringService.tsx
3:    trackError: jest.fn(),
```